scripts/GNN_topk_analysis.py

```diff
@@ -0,0 +1,88 @@
+import argparse
+import logging
+from os import makedirs
+from pathlib import Path
+from shutil import rmtree, copytree
+from typing import Mapping, Sequence
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sn
+import yaml
+
+def parse_in_domain_objects(input_dir: Path) -> Sequence:
+    in_domain_objects = set()
+    for feature_file_path in sorted(input_dir.glob("situation*")):
+        with open(feature_file_path / 'description.yaml', encoding="utf-8") as description_file:
+            description_yaml = yaml.safe_load(description_file)
+            object = description_yaml['language'].rsplit()[-1]
+            in_domain_objects.add(object)
+    return sorted(in_domain_objects)
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Get confusion matrix for GNN decode"
+    )
+    parser.add_argument('--input-dir', type=Path, help="Inpuut curriculum directory to read GNN decodes from")
+    parser.add_argument('--output-dir', type=Path, help="Output directory to write top GNN decode matrix and data to")
+    parser.add_argument(
+        "-f", "--force",
+        action='store_true',
+        required=False,
+        help='Force overwrite of target directory. By default, the script exits with an error if there already exists '
+             'a directory at the target destination.'
+    )
+    args = parser.parse_args()
+
+    logging.basicConfig(level=logging.INFO)
+
+    input_dir: Path = args.input_dir
+    output_dir: Path = args.output_dir
+    force_overwrite: bool = args.force
+
+    if not input_dir.exists():
+        logging.warning("Input directory does not exist")
+        raise FileNotFoundError(str(input_dir))
+
+    if output_dir.is_dir():
+        if not force_overwrite:
+            logging.warning("There already exists a directory in the target location")
+            raise FileExistsError(str(output_dir))
+    else:
+        makedirs(output_dir)
+
+    in_domain_objects = parse_in_domain_objects(input_dir)
+    object_counts = {in_domain_object: 0 for in_domain_object in in_domain_objects}
+    GNN_rank_matrix = [[0 for _ in range(len(in_domain_objects))] for _ in range(len(in_domain_objects))]
```

Comment:
This is better, thank you. Nit: case. I think it's better to stick with the conventional lowercase here.